### PR TITLE
React Native Release 4.0.0

### PR DESCRIPTION
**Pull request by charmian01**
Things to consider before you submit the PR:

* [X] Has `CHANGELOG.md` been updated?
* [X] Are tests passing locally?
* [X] Are the files formatted correctly?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Bumped the versions
Updated TestApp to use latest SDKs
Updated the CHANGELOG.md

### CODE REVIEW

**Comment by zhangcc01** [comment]
@MatkovIvan @russelarms @vvechkanov Please help to check the CHANGELOG.md.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,15 +1,31 @@
...
+* **[Fix]** Fix `NSInvalidArgumentException` when using non-string object as a key in `NSUserDefaults`.
+* **[Fix]** Fix `NSDateFormatter` initialization in a concurrent environment.
+
+### App Center Crashes
+
+#### Android
+
+* **[Feature]** Convert the `saveUncaughtException` method to public and return error log identifier when calling it.
+
+### iOS
```
Comment (DmitriyKirakosyan) [inline CHANGELOG.md:26]: Two `iOS` headers?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,15 +1,31 @@
...
 * **[Breaking change]** Bumping the minimum Android SDK version to 21 API level (Android 5.0), because old Android versions do not support root certificate authority used by App Center and would not get CA certificates updates anymore.
 
 ### iOS
 
-**[Fix]** Fix `Undefined symbols for architecture x86_64` on Xcode 12.
-**[Fix]** Fix naming conflict with iOS 14 private Apple framework.
+* **[Feature]** **[Breaking change]** Make SDK API more Swift-friendly. You can find the API updates in the [swift migration table](https://docs.microsoft.com/appcenter/sdk/getting-started/migration/apple-sdk-update#swift-api-changes).
+* **[Breaking change]** The App Center prefix `MS` in Objective-C is consolidated to `MSAC` for collision prevention. Please follow the [migration guide](https://docs.microsoft.com/appcenter/sdk/getting-started/migration/apple-sdk-update) for adopting the new naming.
+* **[Fix]** Fix `Undefined symbols for architecture x86_64` on Xcode 12.
+* **[Fix]** Fix naming conflict with iOS 14 private Apple framework.
```
Comment (russelarms) [inline CHANGELOG.md:16]:
```suggestion
```
It's a duplicate of 
```
* **[Breaking change]** The App Center prefix `MS` in Objective-C is consolidated to `MSAC` for collision prevention. Please follow the [migration guide](https://docs.microsoft.com/appcenter/sdk/getting-started/migration/apple-sdk-update) for adopting the new naming.
```

Comment (zhangcc01) [reply CHANGELOG.md]: Update in new commit.

**Review by russelarms — APPROVED** [review]

**Pull request merged**